lessons/dotnet6/http-refit/README.md

```diff
@@ -0,0 +1,400 @@
+# Calling third-party REST APIs (.NET 6)
+
+## Goal 🎯
+
+The goal of this lesson is to show how to call third-party REST APIs while using best practices.
+
+Calling REST APIs usually involves the well-known [HttpClient](https://docs.microsoft.com/fr-fr/dotnet/api/system.net.http.httpclient?view=net-6.0) .NET class. However, using this class effectively has always been a challenge as it is designed to be instantiated once and reused troughout the life of your application.
+
+Starting from .NET Core 2.1, the BCL [introduced a set of changes](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests) designed to make `HttpClient` easier to use in a correct and efficient manner.
+
+
+This lesson consists of the following exercises:
+
+|Nr|Exercise
+|-|-
+|0|[Prerequisites](#0-prerequisites)
+|1|[Cleaning up the source code](#1-cleaning-up-the-source-code)
+|2|[Defining a third-party REST API](#2-defining-a-third-party-rest-api)
+|3|[Adding custom API parameters](#3-adding-custom-api-parameters)
+|4|[Homework](#4-homework)
+|5|[More info](#5-more-info)
+
+> 📝 **Tip** - If you're stuck at any point you can have a look at the [source code](../../../src/dotnet6/http-refit/AzFuncUni.Http) in this repository.
+
+---
+
+## 0. Prerequisites
+
+This lesson builds on the [HTTP Trigger (.NET 6)](../http/README.md) lesson. Please, make sure to complete that lesson first.

```

Comment:
Instead of building on the existing HTTP lesson and removing code from it, I suggest we start with a new project, add an HTTP trigger function and include a code snippet (which is the implementation of exercise 4 of the HTTP lesson) here in the markdown to replace the template code of the HTTP trigger function. This is a bit safer, so the two lessons can be independently changed. 

So the prerequisites to this lesson are the same as the HTTP lesson. You can definitely refer to the HTTP lesson if people are unfamiliar with HTTP triggers but don't mention any specifics from that lesson.